### PR TITLE
Make store's fields private

### PR DESCRIPTION
**Pull request by mathetake**
reworkd of #305

### CODE REVIEW

File: internal/wasm/host.go [hunk 1]
```diff
@@ -110,14 +113,95 @@ func (s *Store) NewHostModule(moduleName string, nameToGoFunc map[string]interfa
...
 		}
 	}
-	return &ret, nil
+	return ret, nil
 }
 
-func (s *Store) requireModuleUnused(moduleName string) error {
-	if _, ok := s.hostModules[moduleName]; ok {
-		return fmt.Errorf("module %s has already been exported by this host", moduleName)
+// AddHostFunction exports a function so that it can be imported under the given module and name. If a function already
```
Comment (mathetake) [inline internal/wasm/host.go:119]: the followings are just moved from store.go

File: internal/wasm/store.go [hunk 2]
```diff
@@ -31,23 +31,17 @@ type (
...
 
 		// ModuleInstances holds the instantiated Wasm modules by module name from Instantiate.
-		ModuleInstances map[string]*ModuleInstance
-
-		// hostModules holds host functions by module name from NewHostModule.
-		hostModules map[string]*HostModule
-
-		// ModuleContexts holds default host function call contexts keyed by module name.
-		ModuleContexts map[string]*ModuleContext
+		moduleInstances map[string]*ModuleInstance
```
Comment (mathetake) [inline internal/wasm/store.go:40]: I consolidate moduleContext and hostModule into a part of `ModuleInstance` to ease the management of each objects (under concurrent access)

File: internal/wasm/store.go [hunk 3]
```diff
@@ -31,23 +31,17 @@ type (
 		ctx context.Context
 
 		// Engine is a global context for a Store which is in responsible for compilation and execution of Wasm modules.
-		Engine Engine
+		engine Engine
 
 		// EnabledFeatures are read-only to allow optimizations.
 		EnabledFeatures Features
```
Comment (mathetake) [inline internal/wasm/store.go:37]: I intentionally leave this public as this is read-only as it says :D

**Comment by mathetake** [comment]
almost all here is mechanical!

File: internal/wasm/interpreter/interpreter_test.go [hunk 4]
```diff
@@ -44,31 +44,36 @@ func TestCallEngine_PushFrame_StackOverflow(t *testing.T) {
 }
 
 func TestEngine_Call(t *testing.T) {
+
```
Comment (mathetake) [inline internal/wasm/interpreter/interpreter_test.go:47]:
```suggestion
```

File: internal/wasi/wasi_test.go [hunk 5]
```diff
@@ -954,17 +1028,17 @@ func createFile(t *testing.T, path string, contents []byte) (*memFile, *MemFS) {
...
-// TODO: TestSnapshotPreview1_PathFilestatSetTimes TestSnapshotPreview1_PathFilestatSetTimes_Errors
-// TODO: TestSnapshotPreview1_PathLink TestSnapshotPreview1_PathLink_Errors
-// TODO: TestSnapshotPreview1_PathOpen TestSnapshotPreview1_PathOpen_Errors
-// TODO: TestSnapshotPreview1_PathReadlink TestSnapshotPreview1_PathReadlink_Errors
-// TODO: TestSnapshotPreview1_PathRemoveDirectory TestSnapshotPreview1_PathRemoveDirectory_Errors
-// TODO: TestSnapshotPreview1_PathRename TestSnapshotPreview1_PathRename_Errors
-// TODO: TestSnapshotPreview1_PathSymlink TestSnapshotPreview1_PathSymlink_Errors
-// TODO: TestSnapshotPreview1_PathUnlinkFile TestSnapshotPreview1_PathUnlinkFile_Errors
-// TODO: TestSnapshotPreview1_PollOneoff TestSnapshotPreview1_PollOneoff_Errors
+// // TODO: TestSnapshotPreview1_PathCreateDirectory TestSnapshotPreview1_PathCreateDirectory_Errors
```
Comment (nullpo-head) [inline internal/wasi/wasi_test.go:1031]: `//` is duplicated

**Review by nullpo-head — APPROVED** [review]

**Review by codefromthecrypt — APPROVED** [review]
me like

**Comment by mathetake** [comment]
👯

**Pull request merged**